### ISSUE TITLE
MACS2 fails

### ISSUE THREAD

**Issue by Emiller88** [nf-core/chipseq#284]
Originally reported by Alice Laigle https://nfcore.slack.com/archives/CFP55NZ5G/p1652184721335959
Sarah Muniz Nardeli reported also having an issue with single-end data.

<details> 
  <summary> Error log </summary>

```
# ChIP-seq file = ['WT_H3K27ac.mLb.clN.sorted.bam']
# control file = ['WT_INPUT.mLb.clN.sorted.bam']
# effective genome size = -1.61e+09
# band width = 300
# model fold = [5, 50]
# qvalue cutoff = 5.00e-02
# The maximum gap between significant sites is assigned as the read length/tag size.
# The minimum length of peaks is assigned as the predicted fragment length "d".
# Larger dataset will be scaled towards smaller dataset.
# Range for calculating regional lambda is: 1000 bps and 10000 bps
# Broad region calling is off
# Paired-End mode is off

INFO  @ Thu, 07 Jul 2022 07:41:46: #1 read tag files...
INFO  @ Thu, 07 Jul 2022 07:41:46: #1 read treatment tags...
INFO  @ Thu, 07 Jul 2022 07:41:47:  1000000
INFO  @ Thu, 07 Jul 2022 07:41:49:  2000000
INFO  @ Thu, 07 Jul 2022 07:41:51:  3000000
INFO  @ Thu, 07 Jul 2022 07:41:52:  4000000
INFO  @ Thu, 07 Jul 2022 07:41:54:  5000000
INFO  @ Thu, 07 Jul 2022 07:41:55:  6000000
INFO  @ Thu, 07 Jul 2022 07:41:57:  7000000
INFO  @ Thu, 07 Jul 2022 07:41:59:  8000000
INFO  @ Thu, 07 Jul 2022 07:42:00:  9000000
INFO  @ Thu, 07 Jul 2022 07:42:02:  10000000
INFO  @ Thu, 07 Jul 2022 07:42:03:  11000000
INFO  @ Thu, 07 Jul 2022 07:42:05:  12000000
INFO  @ Thu, 07 Jul 2022 07:42:07:  13000000
INFO  @ Thu, 07 Jul 2022 07:42:08:  14000000
INFO  @ Thu, 07 Jul 2022 07:42:10:  15000000
INFO  @ Thu, 07 Jul 2022 07:42:11:  16000000
INFO  @ Thu, 07 Jul 2022 07:42:13:  17000000
INFO  @ Thu, 07 Jul 2022 07:42:15:  18000000
INFO  @ Thu, 07 Jul 2022 07:42:16:  19000000
INFO  @ Thu, 07 Jul 2022 07:42:18:  20000000
INFO  @ Thu, 07 Jul 2022 07:42:18: 20122646 reads have been read.
INFO  @ Thu, 07 Jul 2022 07:42:18: #1.2 read input tags...
INFO  @ Thu, 07 Jul 2022 07:42:20:  1000000
INFO  @ Thu, 07 Jul 2022 07:42:21:  2000000
INFO  @ Thu, 07 Jul 2022 07:42:23: 2753415 reads have been read.
INFO  @ Thu, 07 Jul 2022 07:42:23: #1 tag size is determined as 47 bps
INFO  @ Thu, 07 Jul 2022 07:42:23: #1 tag size = 47.0
INFO  @ Thu, 07 Jul 2022 07:42:23: #1  total tags in treatment: 20122646
INFO  @ Thu, 07 Jul 2022 07:42:23: #1  total tags in control: 2753415
INFO  @ Thu, 07 Jul 2022 07:42:23: #1 finished!
INFO  @ Thu, 07 Jul 2022 07:42:23: #2 Build Peak Model...
INFO  @ Thu, 07 Jul 2022 07:42:23: #2 looking for paired plus/minus strand peaks...
INFO  @ Thu, 07 Jul 2022 07:42:24: #2 number of paired peaks: 0
WARNING @ Thu, 07 Jul 2022 07:42:24: Too few paired peaks (0) so I can not build the mode
l! Broader your MFOLD range parameter may erase this error. If it still can't build the mod
el, we suggest to use --nomodel and --extsize 147 or other fixed number instead.
WARNING @ Thu, 07 Jul 2022 07:42:24: Process for pairing-model is terminated!
```
</details>

`nf-params.json`
```json
{
  "input": "samplesheet.csv",
  "read_length": 50,
  "outdir": "ming-results",
  "genome": "hg19"
}
```

<details> 
  <summary> samplesheet.csv </summary>

```csv
sample,fastq_1,fastq_2,antibody,control
WT_YAP1,/scratch/applied-genomics/chipseq/ming/SRR1810900.fastq.gz,,YAP1,WT_INPUT
WT_H3K27ac,/scratch/applied-genomics/chipseq/ming/SRR949140.fastq.gz,,H3K27ac,WT_INPUT
WT_INPUT,/scratch/applied-genomics/chipseq/ming/SRR949142.fastq.gz,,,
```
</details

```bash
nextflow run nf-core/chipseq -r 6bf39d5 -profile singularity -params-file nf-params.json
```

Basic fastq download:

```bash
# for YAP1. It takes time, alternatively download fastq directly from EGA https://www.ebi.ac.uk/ega/home
fastq-dump SRR1810900
# for H3K27ac
fastq-dump SRR949140
# for input DNA
fastq-dump SRR949142
```

**Comment by JoseEspinosa** [comment]
The problem here is that macs2 is not able to create the model to estimate fragment size. That's is different from https://github.com/nf-core/chipseq/issues/284#issuecomment-1178025442 since here no output peaks file is produced (in the other case the file was produced but was empty). 
The suggested solution on [this](https://github.com/macs3-project/MACS/issues/390) macs2 issue is to provide `--nomodel` and `--extsize`. I tried `--nomodel --extsize 147`and it works, but that won't be a very suitable solution for the pipeline.

**Comment by JoseEspinosa** [comment]
After a second look, I think something fishy is going on with the `macs_gsize` maps of the `igenomes.config` (or maybe something was already broken before).  I found that the `gsize` provided to the `MACS2_CALLPEAK` is negative, see the command in `.command.sh` when reproducing your run:
```
macs2 \
    callpeak \
    --keep-dup all --broad --broad-cutoff 0.1 \
    --gsize -1610747421 \
    --format BAM \
    --name WT_YAP1 \
    --treatment WT_YAP1.mLb.clN.sorted.bam \
    --control WT_INPUT.mLb.clN.sorted.bam
```

**Comment by Emiller88** [comment]
Ah, I noticed that, but thought it was intentional to avoid some Nextflow thing. Did you try manually flipping that to a positive? Our cluster is down for maintenance until tomorrow now :upside_down_face:

**Comment by JoseEspinosa** [comment]
[This](https://github.com/nf-core/chipseq/pull/285) should fix the issue for you, in the case of the issues mentioned in the slack chat, the problem is different and probably related to my first comment on this issue. In principle, the way to fix it will be to avoid the creation of the model and provide these arguments `--nomodel --extsize 147` via the `modules.config`.

**Comment by Emiller88** [comment]
Using `"macs_gsize": 2684219875`worked! Testing #285 currently with the same run.

**Comment by JoseEspinosa** [comment]
Fixed in #285
